### PR TITLE
User guide initial updates for Jakarta EE 9.1

### PR DESCRIPTION
**Pull request by edbratt**
- Update to Jakarta EE 9.1
- Update CI versions as needed
- Revise the Intro pages to use params. for Release Date and Copyright Dates
- Update Jakarta EE Platform TCK (not parameterized like the others)
- Update copyright dates for 2021
- Add requirements for optional Jakarta XML Binding
- Update Readme and Release Notes as needed

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman

Signed-off-by: Ed Bratt <ed.bratt@oracle.com>

### CODE REVIEW

File: user_guides/jakartaee/src/main/jbake/content/rules-wp.adoc [hunk 1]
```diff
@@ -426,25 +426,26 @@ when processed by the Runtime, must not cause the functional
...
-EE-WP21 Compliance testing for Jakarta EE 9 Web Profile consists of running
-the Jakarta EE 9 Web Profile tests and the following Technology
+EE-WP21 Compliance testing for Jakarta EE 9.1 Web Profile consists of running
+the Jakarta EE 9.1 Web Profile tests and the following Technology
 Compatibility Kits (TCKs):
 
 * Jakarta Contexts and Dependency Injection 3.0
 * Jakarta Dependency Injection 2.0 
 * Jakarta Bean Validation 3.0
+* Jakarta XML Binding 3.0 (If XML Binding is supported)
```
Comment (scottmarlow) [inline user_guides/jakartaee/src/main/jbake/content/rules-wp.adoc:445]: :+1:

File: install/concurrency/docs/index.html [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
-  <title>Welcome to the Concurrency TCK, Version 2.0</title>
-  <style type="text/css">
+  <head>
+    <meta http-equiv="content-type" content="text/html; charset=windows-1252">
+    <!--
+    Copyright (c) 2018, 2020 Oracle and/or its affiliates. All rights reserved.    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v. 2.0, which is available at    http://www.eclipse.org/legal/epl-2.0.    This Source Code may also be made available under the following Secondary
+    Licenses when the conditions for such availability set forth in the    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,    version 2 with the GNU Classpath Exception, which is available at    https://www.gnu.org/software/classpath/license.html.
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0-->
+    <title>Welcome to the Concurrency TCK, Version 2.0</title>
```
Comment (RohitKumarJain) [inline install/concurrency/docs/index.html:10]: As we have updated the TCK version for Standalone TCK's, in that new Concurrency TCK version is updated as 2.0.1, I guess we might need to update this from Version 2.0 to Version 2.0.1

File: install/concurrency/docs/index.html [hunk 3]
```diff
@@ -1,26 +1,14 @@
...
-    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
-
--->
-
-  <title>Welcome to the Concurrency TCK, Version 2.0</title>
-  <style type="text/css">
+  <head>
+    <meta http-equiv="content-type" content="text/html; charset=windows-1252">
+    <!--
+    Copyright (c) 2018, 2020 Oracle and/or its affiliates. All rights reserved.    This program and the accompanying materials are made available under the
```
Comment (RohitKumarJain) [inline install/concurrency/docs/index.html:6]: I believe the Copyright year needs to be updated to 2021.

File: install/connector/docs/ReleaseNotes-JCA-2.0.html [hunk 4]
```diff
@@ -1,130 +1,132 @@
...
+.emphasize  {  font-style: italic; color: #000099}
+.step  { font-weight: bold}
+.note { background-color: #fffff0; border-bottom-width: thin; border-top-width: thin; border-top-style: inset; border-bottom-style: inset; font-size: -10%;}
+th {  background-color: #eeeeee; padding-left: 2pt; padding-right: 2pt;}
+-->
+</style></head>
+  <body>
+    <div align="center">
+      <h1>Jakarta EE Connectors Specification Technology Compatibility Kit,
+        Version 2.0</h1>
```
Comment (RohitKumarJain) [inline install/connector/docs/ReleaseNotes-JCA-2.0.html:29]: For Connectors TCK as well, we have updated the version to 2.0.1

**Review by gurunrao — COMMENTED** [review]
install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html - file name can be updated to 9.1
Also CDI TCK bundle file can be updated to 3.0.1

File: install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html [hunk 5]
```diff
@@ -1,129 +1,146 @@
...
-<hr size="3">
-<small>Copyright &copy; 2020, Oracle and/or its affiliates. All rights reserved.</small>
-</body>
-</html>
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+  <head>
+    <meta http-equiv="content-type" content="text/html; charset=windows-1252">
+    <!--
+    Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
```
Comment (RohitKumarJain) [inline install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html:6]: Copyright year needs to be updated to 2021.

File: user_guides/saaj/src/main/jbake/content/attributes.conf [hunk 6]
```diff
@@ -2,6 +2,8 @@
 :TechnologyShortName: SOAP Attachments
 :LegacyAcronym: SAAJ
 :TechnologyVersion: 2.0
+:ReleaseDate: May 2021
+:CopyrightDates: 2017, 2021
 :TechnologyRI: Jakarta SOAP with Attachments Implementation 2.0.0
```
Comment (RohitKumarJain) [inline user_guides/saaj/src/main/jbake/content/attributes.conf:7]: I believe this should be 2.0.1.

File: install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html [hunk 7]
```diff
@@ -1,129 +1,146 @@
...
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v. 2.0, which is available at    http://www.eclipse.org/legal/epl-2.0.
+    This Source Code may also be made available under the following Secondary
+    Licenses when the conditions for such availability set forth in the    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+    version 2 with the GNU Classpath Exception, which is available at    https://www.gnu.org/software/classpath/license.html.
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+-->
+    <title>Release Notes for Jakarta EE 9 Platform TCK</title>
```
Comment (RohitKumarJain) [inline install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html:14]: we need to update to Jakarta EE 9.1

File: install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html [hunk 8]
```diff
@@ -1,129 +1,146 @@
...
+    </div>
+    <hr>
+    <table title="Note" border="0">
+      <tbody>
+        <tr valign="top">
+          <td><strong>Note:</strong></td>
+          <td><em>This is <strong>the initial version </strong> of the Jakarta
+              EE 9.1 TCK. The Release Notes file may be updated via the Jakarta
+              EE Platform Specification Web Site: <a href="https://jakarta.ee/specifications/platform/9/">https://jakarta.ee/specifications/platform/9/</a>
+              contains the most current information about the Jakarta EE 9
```
Comment (RohitKumarJain) [inline install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html:46]: an update might require for Jakarta EE 9 to Jakarta EE 9.1

File: install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html [hunk 9]
```diff
@@ -1,129 +1,146 @@
...
+      <li>XML Webservices version 3.0 (Optional)</li>
+      <li>SOAP version 2.0 (Optional)</li>
+    </ul>
+    <hr>
+    <h2>Status Information</h2>
+    <h3>Exclude List</h3>
+    <blockquote>
+      <p>You can find the exclude list in the CTS bundle in <code>&lt;TS_HOME&gt;/bin/ts.jtx</code>.
+        New versions of the exclude list are released and the TCK bundle will be
+        updated. Be sure to check for updates at the <em> Jakarta EE 9 Platform
```
Comment (RohitKumarJain) [inline install/jakartaee/docs/ReleaseNotes-JakartaEETCK-9.0.html:112]: an update might require for Jakarta EE 9 to Jakarta EE 9.1

File: install/jsf/docs/ReleaseNotes.html [hunk 10]
```diff
@@ -99,98 +92,107 @@
...
+    <p>Refer to the <cite>Jakarta EE Server Faces TCK 3.0 User's Guide</cite> (<a
+
+        href="./html-usersguide/title.html" title="Jakarta EE Server Faces TCK 3.0 User's Guide (HTML)">HMTL</a>,
+      <a href="./pdf-usersguide/Jakarta-Server-Faces-TCK-Users-Guide.pdf" title="Jakarta EE Server Faces TCK 3.0 User's Guide (PDF)">PDF</a>)
+      for platform compatibility requirements and notes.</p>
+    <hr>
+    <h2>Compatible Implementation</h2>
+    <p>A compatible implementation for this release is Eclipse Mojarra which
+      includes Jakarta EE Server Faces 3.0. To run the TCK, you will need to use
+      a Web container or a Jakarta EE 9 compatible server that has integrated
```
Comment (RohitKumarJain) [inline install/jsf/docs/ReleaseNotes.html:158]: an update might require for Jakarta EE 9 to Jakarta EE 9.1

File: install/jsf/docs/ReleaseNotes.html [hunk 11]
```diff
@@ -99,98 +92,107 @@
...
+      Mojarra 3.0. For the current Jakarta EE 9 server, you will need to perform
```
Comment (RohitKumarJain) [inline install/jsf/docs/ReleaseNotes.html:159]: an update might require for Jakarta EE 9 to Jakarta EE 9.1

File: install/concurrency/docs/index.html (hunk 2 above)
Comment (RohitKumarJain) [reply install/concurrency/docs/index.html:10]: While reviewing the rest of the user guide changes, I see the version is limited up to two digits, for example, 2.0 or 3.0, etc. So I believe this is what it supposed to be. Still, I would like to hear more from experts.

Comment (scottmarlow) [reply install/concurrency/docs/index.html:10]: The Spec version for Concurrency is 2.0 as is shown on https://jakarta.ee/specifications/concurrency.  Any two digit version number such as `2.0` is referring to the SPEC version, not the TCK version.  

In summary, the individual SPEC versions should not be changed for Jakarta EE 9.1, as the SPECs are not being released again.  So, please do update Concurrency TCK 2.0.0 references to the new TCK version, but do not change the Concurrency SPEC version which should stay 2.0.

Comment (edbratt) [reply install/concurrency/docs/index.html:10]: I removed the version number from the title element to match the others.

File: install/connector/docs/ReleaseNotes-JCA-2.0.html (hunk 4 above)
Comment (edbratt) [reply install/connector/docs/ReleaseNotes-JCA-2.0.html:29]: I don't intend to update the micro-version changes.

**Review by alwin-joseph — APPROVED** [review]
I verified the JakartaEE TCK users guide by generating the same in CI with this change - https://ci.eclipse.org/jakartaee-tck/job/buildUG/22/artifact/user_guides/jakartaee/target/generated-docs/Jakarta-EE-TCK-Users-Guide.pdf , same LGTM.

**Comment by edbratt** [comment]
> I verified the JakartaEE TCK users guide by generating the same in CI with this change - https://ci.eclipse.org/jakartaee-tck/job/buildUG/22/artifact/user_guides/jakartaee/target/generated-docs/Jakarta-EE-TCK-Users-Guide.pdf , same LGTM.

If you follow that link and look on page 9, you should see a graphic image on page 9 but it isn't present (section 1.2.3, image 1.1). I believe, with Lukas' fix, this image will now render properly in the PDF. I'm ready to merge this as soon a folks indicate they are happy with my updates from today (or if further changes are needed, please comment.)

**Review by RohitKumarJain — APPROVED** [review]
LGTM !!!

**Review by gurunrao — APPROVED** [review]

**Comment by alwin-joseph** [comment]
> > I verified the JakartaEE TCK users guide by generating the same in CI with this change - https://ci.eclipse.org/jakartaee-tck/job/buildUG/22/artifact/user_guides/jakartaee/target/generated-docs/Jakarta-EE-TCK-Users-Guide.pdf , same LGTM.
> 
> If you follow that link and look on page 9, you should see a graphic image on page 9 but it isn't present (section 1.2.3, image 1.1). I believe, with Lukas' fix, this image will now render properly in the PDF. I'm ready to merge this as soon a folks indicate they are happy with my updates from today (or if further changes are needed, please comment.)
> 
Thanks. I missed to see that.

**Comment by scottmarlow** [comment]
Thanks @edbratt  for the heroic User Guide update!  Very much appreciated!

**Pull request merged**